vernac/mltop.ml

```diff
@@ -287,16 +365,15 @@ let add_known_module m = add_known_module m
 let loaded_modules = ref []
 let get_loaded_modules () = List.rev !loaded_modules
 let add_loaded_module md =
+  (* XXX: This if is suspicious, we should guard in the caller *)
```

Comment:
I believe this is there becase module unloading is not a thing in OCaml, so even if you undo a `Declare ML Module` both this ref and the actual ocaml code stays unchanged. Where would you move this logic?

Reply:
I mean this logic could be put in trigger_ml_object, but indeed, we can do some further cleanup in another PR, maybe I remove the comment if that's confusing.

Reply:
ok